cms/static/cms/js/modules/cms.modal.js

```diff
@@ -50,6 +50,9 @@ var CMS = window.CMS || {};
                 // event emitter
                 this._setupEventEmitter();
 
+                // reference messaging system
+                this.messages = new CMS.Message();
```

Comment:
doing it separately means that you're creating an instance for _each_ plugin which in turn means higher memory footprint etc.